22.md

```diff
@@ -0,0 +1,65 @@
+# NIP-22 
+
+## Smart Widgets
+`draft` `optional`
+
+Smart Widgets are customizable, interactive components that can be embedded within Nostr events and allow to create rich, dynamic content. This proposal outlines the structure, implementation, and potential use cases of Smart Widgets.
+
+## Motivation
+Traditional static content limits user interaction and engagement. Smart Widgets provide a way to embed dynamic components, such as buttons, images, and polls, directly into Nostr events, enabling more interactive and responsive content.
```

Comment:
To support such a thing we can simply use HTML. put some HTML5 in `.content` and then the client can parse it like a webpage or built-in Telegram browser that we can see a lot nowadays.

But overall about parsing HTML, we have a closed issue which I suggest you to check.

- #1442

It can bring the potential of embedding games/web pages on events as well. A cool idea. But based on the above issue we have reasons to avoid that. IMO we can simply support this on some specific clients with no standards. If they find important use cases we can make them standard.

Reply:
This pr is all about letting clients decide what they need: `html` or `html5` or `xml` or `svg` or `json` or pretty much whatever. For example, if yakihonnes wants stringified json they can use it without forcing everyone to accept it as a standard.

Regarding that issue, I don’t understand why whould you need a separate kind for each, instead ordering clients to redirect in case of no support for a specific mime type by using `naddr` and the `client` used for creating smart widget.

Beyond that, I completely agree with you.

Reply:
Few thoughts to clarify my view and avoid misundertanding. IMO this pr is different from #811 . Adding scripts is unnecessary for this specific pr NO SCRIPTS. Adding games and webpages with mixed formats is too much. The idea is great, no doubt, but smart widgets are by no mean a decentralized web hosting on Nostr. Smart widgets don't need variables like css3 or javascript etc. These are just dynamic interactive images that user can click on. Forms that user can submit and other parsable content.